interfaces/IMonitor.h

```diff
@@ -0,0 +1,89 @@
+            uint64_t average /* @brief Average of all measurements */;
+            uint64_t last /* @brief Last measured value */;
+        };
+        struct Statistics {
+            Measurement resident /* @brief Resident memory measurement */;
+            Measurement allocated /* @brief Allocated memory measurement */;
+            Measurement shared /* @brief Shared memory measurement */;
+            Measurement process /* @brief Processes measurement */;
+            bool operational /* @brief Whether the service is up and running */;
+            uint32_t count /* @brief Number of measurements */;
+        };
+        struct RestartInfo {
+            uint8_t limit /* @brief Maximum number or restarts to be attempted */;
+            uint16_t window /* @brief Time period (in seconds) within which failures must happen for the limit to be considered crossed */;
+        };
+
+        using IStringIterator = RPC::IIteratorType<string, RPC::ID_STRINGITERATOR>;
+
+        // @event
+        struct EXTERNAL INotification : virtual public Core::IUnknown {
+            enum { ID = ID_MONITOR_NOTIFICATION };
+            enum action : uint8_t {
+                DEACTIVATE,
+                ACTIVATE,
+                RESTARTING_IS_STOPPED,
+            };
+            // @brief Signals an action taken by the Monitor
+            // @param callsign: Callsign of the service the Monitor acted upon
+            // @param action: The action executed by the Monitor on a service. One of: Activate, Deactivate, StoppedRestarting
+            // @param reason: A message describing the reason the action was taken
+            virtual void Actionconst string& callsign, const action value, const string& reason) = 0;
+        };
+
+        // Pushing notifications to interested sinks
+        virtual Core::hresult Register(INotification* notification) = 0;
+        virtual Core::hresult Unregister(INotification* notification) = 0;
+
+        // @property
+        // @brief RestartLimits: Set/Get new restart limits for a plugin
+        // @param callsign: Callsign of the plugin
+        // @param params: Restart Information
+        virtual Core::hresult RestartLimits(const string& callsign /* @index */, const RestartInfo& params) = 0;
+        virtual Core::hresult RestartLimits(const string& callsign /* @index */, RestartInfo& params /* @out */) const = 0;
+
+        // @brief ResetStats: Resets memory for a single plugin watched by the Monitor
+        virtual Core::hresult Reset(const string& callsign) = 0;
+
+        // @property
+        // @brief Observables: List of plugins watched by the Monitor
+        virtual Core::hresult Observables(IStringIterator*& observables /* @out */) const = 0;
+        // @property
+        // @brief Statistics: The memory and process statistics either for a given plugin
+        virtual Core::hresult StatisticsInfo(const string& callsign /* @index */, Statistics& statistics /* @out */) const = 0;
```

Comment:
I suppose everything is information, so *Info feels redundant, why not have it just Statistics() ?